politeiawww/mail/mock_database_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package mail
+
+import (
+	"github.com/decred/politeia/politeiawww/user"
+	"github.com/google/uuid"
+)
+
+// Ensure, that DatabaseMock does implement user.Database.
+var _ user.Database = &DatabaseMock{}
+
+// DatabaseMock is a mock implementation of user.Database.
+type DatabaseMock struct {
```

Comment:
I'm not a fan of these mock implementations.  It creates too much overhead for writing tests when you have to define the function in every test.  The way this should be done is the create a `user/testdb` implementation that implements the `user.Database` interface by saving and retreiving from memory.  It makes it much easier to write and read tests when written like this.

Same for the mock mailer.

Reply:
Sorry, but I don't agree with this point (and there are a lot of people in Golang community who don't agree as well, those who are using https://github.com/matryer/moq being an obvious example).

I think it is a waste of time rewriting the way mocking is done in this test, just because you prefer using other approaches, so I won't do it without objective reasons to.